src/js/cursor.ts

```diff
@@ -0,0 +1,159 @@
+import { KEYS } from "./constants.js";
+import { createCursorSpan } from "./utils.js";
+
+export class Cursor {
+  position: number;
+  element: HTMLSpanElement;
+
+  constructor() {
+    this.position = 0;
+    this.element = document.getElementById("cursor") as HTMLSpanElement;
+  }
+
+  private remove() {
+    document.getElementById("cursor")!.remove();
+  }
+
+  private place(input: null | HTMLSpanElement) {
+    let content = input?.textContent;
+    this.element = createCursorSpan();
+    input!.innerHTML =
+      content?.slice(0, this.position) +
+      this.element.outerHTML +
+      content?.slice(this.position);
+  }
+
+  resetPosition() {
+    this.position = 0;
+  }
+
+  setPosition(position: number, input: null | HTMLSpanElement) {
+    if (position == 0) {
+      this.position = 0;
+      return;
+    }
+    if (position > 0 && position <= input!.textContent!.length) {
+      this.position = position;
+    }
+  }
+
+  addTextContent(input: null | HTMLSpanElement, textContent: string) {
+    this.remove();
```

Comment:
For `remove` and `place` we should consider using [decorators](https://www.geeksforgeeks.org/what-are-decorators-and-how-are-they-used-in-javascript/).

Reply:
**Currently could not be implemented**